bindings/dart/lib/src/web/ffi/library.dart

```diff
@@ -16,9 +18,26 @@ external _Promise<Wasm> _init([
   dynamic module_or_path,
 ]);
 
+@JS('xayn_ai_ffi_wasm.initThreadPool')
+external _Promise<void> _initThreadPool(int numberOfThreads);
+
+@JS('xayn_ai_ffi_wasm.needs_parallel_init')
+external bool _needsParallelInit();
+
 /// Initializes the wasm module.
 ///
 /// If `moduleOrPath` is a `RequestInfo` or `URL`, makes a request and
 /// for everything else, calls `WebAssembly.instantiate` directly.
-Future<Wasm> init([dynamic moduleOrPath]) async =>
-    promiseToFuture(_init(moduleOrPath));
+Future<Wasm> init([dynamic moduleOrPath]) async {
+  final wasm = await promiseToFuture<Wasm>(_init(moduleOrPath));
+  if (_needsParallelInit()) {
```

Comment:
Can't we just have an `_initThreadPool` that does nothing in the wasm that does not support parallelization?

Reply:
Yes I will change it.